AWSPinpoint/AWSPinpointNotificationManager.m

```diff
@@ -260,6 +269,54 @@ - (void)addApplicationStateAttributeToEvent:(AWSPinpointEvent *) event
     }
 }
 
+- (AWSPinpointEvent*)buildEventFromUserInfo:(NSDictionary *) userInfo
+                         withPushActionType:(AWSPinpointPushActionType) pushActionType {
+    NSString *eventTypePrefix = [self getEventTypePrefixFromUserInfo:userInfo];
+    NSString *eventTypeSuffix = [self getEventTypeSuffixFromPushActionType:pushActionType];
+    if (!eventTypePrefix || !eventTypeSuffix) {
+        return nil;
+    }
+    NSString *eventType = [NSString stringWithFormat: @"_%@.%@", eventTypePrefix, eventTypeSuffix];
+
+    AWSPinpointEvent *pushNotificationEvent  = [self.context.analyticsClient createEventWithEventType:eventType];
+
+    [self addApplicationStateAttributeToEvent:pushNotificationEvent
+                         withApplicationState:[[UIApplication sharedApplication] applicationState]];
+    return pushNotificationEvent;
+}
+
+- (NSString*)getEventTypePrefixFromUserInfo:(NSDictionary *) userInfo {
+    if ([AWSPinpointNotificationManager validPinpointPushForNotification:userInfo]) {
+        NSDictionary *pinpointData = userInfo[AWSDataKey][AWSPinpointKey];
+        if (pinpointData[AWSPinpointCampaignKey]) {
+            return AWSPinpointCampaignKey;
+        } else if (pinpointData[AWSPinpointJourneyKey]) {
+            return AWSPinpointJourneyKey;
+        }
+    }
+    AWSDDLogError(@"Cannot determine event type from Push payload");
+    return nil;
+}
+
+- (NSString*)getEventTypeSuffixFromPushActionType:(AWSPinpointPushActionType) pushActionType {
+    NSString *eventType;
+    switch (pushActionType) {
+        case AWSPinpointPushActionTypeOpened:
+            eventType = AWSEventTypeOpened;
+            break;
+        case AWSPinpointPushActionTypeReceivedForeground:
+            eventType = AWSEventTypeReceivedForeground;
+            break;
+        case AWSPinpointPushActionTypeReceivedBackground:
+            eventType = AWSEventTypeReceivedBackground;
+            break;
+        case AWSPinpointPushActionTypeUnknown:
+            eventType = nil;
```

Comment:
no need for `eventType = nil` due to ARC.  eventType is already nil

Reply:
I generally prefer to make all cases explicit so that the value is clear.
But I will update.